lib/Controller/FolderController.php

```diff
@@ -164,12 +165,28 @@ public function getFolder(int $id): DataResponse {
 	private function checkFolderExists(int $id): ?DataResponse {
 		$storageId = $this->getRootFolderStorageId();
 		if ($storageId === null) {
-			return new DataResponse([], Http::STATUS_NOT_FOUND);
+			throw new OCSNotFoundException('Groupfolder not found');
 		}
 
 		$folder = $this->manager->getFolder($id, $storageId);
 		if ($folder === null) {
-			return new DataResponse([], Http::STATUS_NOT_FOUND);
+			throw new OCSNotFoundException('Groupfolder not found');
+		}
+
+		return null;
+	}
+
+	private function checkMountPointExists(string $mountpoint): ?DataResponse {
+		$storageId = $this->getRootFolderStorageId();
+		if ($storageId === null) {
+			throw new OCSNotFoundException('Groupfolder not found');
+		}
+
```

Comment:
why is this here?

Reply:
Because psalm isn't smart enough to understand the check above :facepalm:

Reply:
Then you should rather suppress it than bending your code for psalm to be happy.

Reply:
With something like `@var` ?

Reply:
Yeah or just `@psalm-supress`

Reply:
Not sure if really cleaner, but here it is: https://github.com/nextcloud/groupfolders/pull/3802